File: src/app/models/track.ts
import {EntryModel} from "./entry";

export interface TrackModel {
  id: number;
  name: string;
  location: string;
  length: number;
  entryList: EntryModel[];
}

export type TrackModelPost = Partial<TrackModel> & {id: number}

// export class Track{
//   private _id: number;
//   private _name: string;
//   private _location: string;
//   private _length: number;
//   private _entryList: Entry[] | null;
//
//
//   constructor(id: number, name: string, location: string, length: number, entryList: [] | null) {
//     this._id = id;
//     this._name = name;
//     this._location = location;
//     this._length = length;
//     this._entryList = entryList;
//   }
//
//   get id(): number {
//     return this._id;
//   }
//
//   set id(value: number) {
//     this._id = value;
//   }
//
//   get name(): string {
//     return this._name;
//   }
//
//   set name(value: string) {
//     this._name = value;
//   }
//
//   get entries(): Entry[] | null{
//     return this._entryList;
//   }
//
//   set entries(value: Entry[] | null) {
//     this._entryList = value;
//   }
//
//   get location(): string {
//     return this._location;
//   }
//
//   set location(value: string) {
//     this._location = value;
//   }
//
//   get length(): number {
//     return this._length;
//   }
//
//   set length(value: number) {
//     this._length = value;
//   }
// }

File: backend/src/main/java/com/example/backend/models/Track.java
package com.example.backend.models;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonManagedReference;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.List;

@Entity
public class Track {
    @Id
    @Column(name = "track_id")
    @GeneratedValue(strategy = GenerationType.AUTO)
    private int id;

    @Column(name = "name")
    private String name;

    @Column(name = "location")
    private String location;

    @Column(name = "length")
    private double length;

//    @OneToMany(mappedBy = "track")
//    @JsonBackReference
//    private List<Round> roundList = new ArrayList<>();

    @OneToMany(mappedBy = "track")
    @JsonManagedReference(value = "entry-track")
    private List<Entry> entryList = new ArrayList<>();

  public Track(int id, String name, String location, double length, List<Entry> entryList) {
    this.id = id;
    this.name = name;
    this.location = location;
    this.length = length;
    this.entryList = entryList;
  }

  public Track() {
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getLocation() {
        return location;
    }

    public void setLocation(String location) {
        this.location = location;
    }

    public double getLength() {
        return length;
    }

    public void setLength(double length) {
        this.length = length;
    }

    public List<Entry> getEntryList() {
        return entryList;
    }

    public void setEntryList(List<Entry> entryList) {
        this.entryList = entryList;
    }

//  public List<Round> getRoundList() {
//    return roundList;
//  }
//
//  public void setRoundList(List<Round> roundList) {
//    this.roundList = roundList;
//  }
}
